From 52010ff3a302cc124c85161d785a2a3bd77b2675 Mon Sep 17 00:00:00 2001
Message-Id: <52010ff3a302cc124c85161d785a2a3bd77b2675.1437120032.git.chang-joon.lee@intel.com>
From: N Tajunnisha <tajunnisha.n@intel.com>
Date: Fri, 3 Jul 2015 15:53:56 +0530
Subject: [PATCH] Revert "REVERTME [VPG]: drm/i915: Adding X-Power PMIC
 support"

This reverts commit 9b7f1f7ba9cc484d7d6a732b62d6d076e564ed3c.
GOP added the support for both Xpower and T1PMIC through vbt.
Hence reverting the hardcoded Xpower PMIC values in driver.

Change-Id: I865cb977c8771538493efaa3051b0179d0c4a787
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-39448
Signed-off-by: N Tajunnisha <tajunnisha.n@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h            |  8 --------
 drivers/gpu/drm/i915/intel_dsi_panel_vbt.c | 29 +++++++----------------------
 2 files changed, 7 insertions(+), 30 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index a559a0a..e95af3a 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -1799,11 +1799,6 @@ enum punit_power_well {
 #define   FBC_REND_NUKE		(1<<2)
 #define   FBC_REND_CACHE_CLEAN	(1<<1)
 
-/* PMIC Registers */
-#define XPOWER_PMIC_PANEL_POWER_CTRL_REG 0x12
-#define XPOWER_PMIC_PANEL_POWER_CTRL_DATAMASK (1 << 6)
-#define XPOWER_PMIC_PANEL_POWER_ON_DATA (1 << 6)
-
 /*
  * GPIO regs
  */
@@ -7187,7 +7182,4 @@ enum punit_power_well {
 
 #define GEN8_RC6_WA_BB	    0x2058
 
-/* CHT_CR Related */
-#define CHT_CR_REVISION 0x22
-
 #endif /* _I915_REG_H_ */
diff --git a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
index 4ecc9e2..87beefa 100644
--- a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
+++ b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
@@ -544,7 +544,6 @@ static u8 *mipi_exec_spi(struct intel_dsi *intel_dsi, u8 *data)
 
 static u8 *mipi_exec_pmic(struct intel_dsi *intel_dsi, u8 *data)
 {
-	struct drm_device *dev = intel_dsi->base.base.dev;
 	u32 register_address, register_data;
 	int data_mask, tmp;
 	int ret;
@@ -553,28 +552,14 @@ static u8 *mipi_exec_pmic(struct intel_dsi *intel_dsi, u8 *data)
 	 * First 3 bytes are not relevant for Linux.
 	 * Skipping the data field by 3 bytes to get
 	 * the PMIC register Address.
-	 * For CHT-CR, Xpower PMIC is POR for Android
-	 * but GOP supports a different PMIC for windows.
-	 * So, hardcoding the PMIC register and values for now
-	 * until GOP supports XPower PMIC.
 	 */
-	if (IS_CHERRYVIEW(dev) && dev->pdev->revision == CHT_CR_REVISION) {
-		data += 7;
-		register_address = XPOWER_PMIC_PANEL_POWER_CTRL_REG;
-		register_data = *((u32 *)data);
-		data_mask = XPOWER_PMIC_PANEL_POWER_CTRL_DATAMASK;
-		if (register_data != 0)
-			register_data = XPOWER_PMIC_PANEL_POWER_ON_DATA;
-		data += 8;
-	} else {
-		data += 3;
-		register_address = *((u32 *)data);
-		data += 4;
-		register_data = *((u32 *)data);
-		data += 4;
-		data_mask = *((u32 *)data);
-		data += 4;
-	}
+	data += 3;
+	register_address = *((u32 *)data);
+	data += 4;
+	register_data = *((u32 *)data);
+	data += 4;
+	data_mask = *((u32 *)data);
+	data += 4;
 
 	tmp = dsi_soc_pmic_readb(register_address);
 	if (tmp < 0)
-- 
1.9.1

